relay/pkg/block/forwarder.go

```diff
@@ -0,0 +1,138 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+	"github.com/keep-network/tbtc/relay/pkg/chain"
+)
+
+const (
+	// Size of the headers queue.
+	headersQueueSize = 50
+
+	// Maximum size of processed headers batch.
+	headersBatchSize = 5
+
+	// Maximum time for which the pulling process will wait for a single header
+	// to be delivered by the headers queue.
+	headerTimeout = 1 * time.Second
+
+	// Block duration of a Bitcoin difficulty epoch.
+	difficultyEpochDuration = 2016
+
+	// Duration for which the forwarder should rest after performing
+	// a push action.
+	forwarderPushingSleepTime = 45 * time.Second
+)
+
+var logger = log.Logger("relay-block-forwarder")
+
+// Forwarder takes blocks from the Bitcoin chain and forwards them to the
+// given host chain.
+type Forwarder struct {
+	btcChain  btc.Handle
+	hostChain chain.Handle
+
+	processedHeaders int
+
+	headersQueue chan *btc.Header
+	errChan      chan error
+
+	loopExitHandler func()
+}
+
+// RunForwarder creates an instance of the block forwarder and runs its
+// processing loops. The lifecycle of the forwarder can be managed using the
+// passed context.
+func RunForwarder(
```

Comment:
If we decide to rename the package to `relay`, I propose we name it `Start` and make it clear in the docs that the relay exits on a first error. This will explain the logic in `runForwarderControlLoop` in `node.go`.

Reply:
We'll make it `StartRelay` as per https://github.com/keep-network/tbtc/pull/767#discussion_r603913978